api/buckets/service.go

```diff
@@ -2084,26 +2147,54 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		}
 	}
 
+	ba, err := s.Collections.BucketArchives.GetOrCreate(ctx, req.Key)
+	if err != nil {
+		return nil, fmt.Errorf("getting bucket archive data: %s", err)
+	}
+
+	var archiveConfig *mdb.ArchiveConfig
+	if req.ArchiveConfig != nil {
+		archiveConfig = fromPbArchiveConfig(req.ArchiveConfig)
+	} else if ba.DefaultArchiveConfig != nil {
+		archiveConfig = ba.DefaultArchiveConfig
+	} else {
+		// The queried BucketArchive was created before we added .DefaultArchiveConfig so just use the default default
+		// and defer saving the default until the user calls DefaultArchiveConfic or SetDefaultArchiveConfig.
+		archiveConfig = &defaultDefaultArchiveConfig
```

Comment:
Or just use the default default that is specified in code at the top of this file.